### PR TITLE
Check for intel-openmp DLLs before importing torch

### LINKED ISSUE #35803: FileNotFoundError - caffe2_detectron_ops.dll on Windows source build if Python 3.8 used
Hi, I'm getting this dll import errors when I'm trying to import torch since 1.4 if I use Python 3.8 for build. The error below is from source build in a virtualenv that I tried today. I can build and import it just fine if I use Python 3.7

```
>>> import torch
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "F:\pythonapps\pytorch-test\env\lib\site-packages\torch\__init__.py", line 77, in <module>
    ctypes.CDLL(dll)
  File "C:\Users\onurc\AppData\Local\Programs\Python\Python38\lib\ctypes\__init__.py", line 373, in __init__
    self._handle = _dlopen(self._name, mode)
FileNotFoundError: Could not find module 'F:\pythonapps\pytorch-test\env\lib\site-packages\torch\lib\caffe2_detectron_ops.dll' (or one of its dependencies). Try using the full path with constructor syntax.
```

**Environment**
- PyTorch Version (e.g., 1.0): master
- OS (e.g., Linux): Windows 10
- How you installed PyTorch (conda, pip, source): source
- Build command you used (if compiling from source): python setup.py install
- Python version: 3.8.2
- CUDA/cuDNN version: 10.2
- GPU models and configuration:
- Any other relevant information: VS 2017

### PR DESCRIPTION
**Pull request by peterjc123**
Fixes https://github.com/pytorch/pytorch/issues/35803#issuecomment-682356457

### CODE REVIEW

File: torch/__init__.py [hunk 1]
```diff
@@ -126,6 +126,26 @@
...
+        if cuda_version not in ('9.2', '10.0'):
+            load_library_with_flags('vcruntime140_1.dll')
+    except OSError:
+        print('''Microsoft Visual C++ Redistributable is not installed, which may lead to the DLL load failure.
+                 It can be downloaded at https://aka.ms/vs/16/release/vc_redist.x64.exe''')
+
+    try:
+        load_library_with_flags('libiomp5md.dll')
+    except OSError:
+        print('''Intel OpenMP is not installed, which may lead to the DLL load failure.
```
Comment (nbcsm) [inline torch/__init__.py:141]: libiomp5md.dll is part of pytorch wheel, why user still need to install Intel OpenMP?

**Comment by ezyang** [comment]
@malfet I wonder if we can get codecov working on the Windows build; that might actually be pretty useful application of coverage on these very fiddly bits.

File: torch/__init__.py (hunk 1 above)
Comment (ezyang) [reply torch/__init__.py:141]: I guess, people aren't necessarily always going to be using our prebuilt Wheels. But that leads me to another question @peterjc123: is Intel OpenMP the *only* openmp configuration we support on Windows?

**Comment by malfet** [comment]
> @malfet I wonder if we can get codecov working on the Windows build; that might actually be pretty useful application of coverage on these very fiddly bits.



Filed https://github.com/pytorch/pytorch/issues/43897

File: torch/__init__.py (hunk 1 above)
Comment (peterjc123) [reply torch/__init__.py:141]: I think it is the root cause for the issue https://github.com/pytorch/pytorch/issues/35803#issuecomment-682356457, but it doesn't seem to be that case. But right now, I'm trying to do sth. like @ezyang described.
As for OpenMP, actually we support VCOMP (MSVC Bulitin) and Intel OpenMP, but someone claims that PyTorch fails to import if `detectron` is built with VCOMP. We may look into that at some time.

Comment (ezyang) [reply torch/__init__.py:141]: OK. So just to double check, this doesn't seem to fix #35803, so we should wait on it some more?

Comment (peterjc123) [reply torch/__init__.py:141]: @ezyang Let's wait for a while. This PR isn't that urgent, I think.

Comment (peterjc123) [reply torch/__init__.py:141]: @ezyang But according to the user's feedback in that issue, it seems that either `asmjit` or `fbgemm` is using VCOMP. I think we should use a unified choice for OMP (->Intel OMP).

**Pull request closed without merging**